pilot/pkg/networking/core/tracing.go

```diff
@@ -462,6 +495,116 @@ func otelLightStepConfig(clusterName, hostname, accessToken string) (*anypb.Any,
 	return anypb.New(dc)
 }
 
+func configureDynatraceSampler(hostname, cluster string,
+	otelProvider *meshconfig.MeshConfig_ExtensionProvider_OpenTelemetryTracingProvider,
+	sampler *meshconfig.MeshConfig_ExtensionProvider_OpenTelemetryTracingProvider_DynatraceSampler_,
+	pushCtx *model.PushContext) (*core.TypedExtensionConfig, error) {
+
+	dsc := &otelsamplers.DynatraceSamplerConfig{
+		Tenant:             sampler.DynatraceSampler.GetTenant(),
+		ClusterId:          sampler.DynatraceSampler.GetClusterId(),
+		RootSpansPerMinute: sampler.DynatraceSampler.GetRootSpansPerMinute(),
+	}
+
+	if sampler.DynatraceSampler.HttpService == nil {
+		// The Dynatrace sampler can re-use the same Dynatrace service/cluster/auth header
+		// as configured for the HTTP Exporter. In this case users
+		// can achieve a much smaller/simpler config in Istio.
+		var header *meshconfig.MeshConfig_ExtensionProvider_HttpHeader
+
+		// Re-use the Dynatrace API Host + Token from the HTTP exporter
+		httpService := otelProvider.GetHttp()
+		if httpService == nil {
+			return nil, fmt.Errorf("dynatrace sampler could not get http settings. considering using dynatrace_api field")
+		}
+
+		uri, err := url.JoinPath(hostname, "api/v2/samplingConfiguration")
+		if err != nil {
+			return nil, fmt.Errorf("could not parse dynatrace adaptative sampling endpoint %v", err)
+		}
+
+		// Extracts the Authorization header from the HTTP exporter headers
```

Comment:
Can't we just set all the headers which are set for the HTTP exporter?

Reply:
I guess that makes sense. I wanted to have a more strict validation but I think it doesn't make much sense. Also makes the code simpler :) I will change

Reply:
Done [add7cfe](https://github.com/dynatrace-oss-contrib/istio/pull/2/commits/add7cfe82dfd608f5d2a23aff0b71fbff7411331)